cpp/zeder_to_zotero_importer.cc

```diff
@@ -0,0 +1,432 @@
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include <iostream>
+#include <unordered_set>
+#include <cstdio>
+#include <cstdlib>
+#include "FileUtil.h"
+#include "IniFile.h"
+#include "util.h"
+#include "ZoteroHarvesterConfig.h"
+#include "ZoteroHarvesterUtil.h"
+#include "ZoteroHarvesterZederInterop.h"
+
+
+namespace {
+
+
+using namespace ZoteroHarvester;
+
+
+[[noreturn]] void Usage() {
+    std::cerr << "Usage: " << ::progname << " [options] config_file_path mode zeder_flavour zeder_ids fields_to_update\n"
+              << "\n"
+              << "\tOptions:\n"
+              << "\t[--min-log-level=log_level]     Possible log levels are ERROR, WARNING (default), INFO and DEBUG\n"
+              << "\t[--overwrite-on-import]         Overwrite existing entries/sections when performing an import\n"
+              << "\n"
+              << "\tconfig_file_path                Path to the Zotero Harvester config file\n"
+              << "\tmode                            Either IMPORT or UPDATE\n"
+              << "\tzeder_flavour                   Either IXTHEO or KRIMDOK\n"
+              << "\tzeder_ids                       Comma-separated list of Zeder entry IDs to import/update.\n"
+              << "\t                                Special-case for updating: Use '*' to update all entries found in the config that belong to the Zeder flavour\n"
+              << "\tfields_to_update                Comma-separated list of the following fields to update: \n"
+              << "\t                                \tONLINE_PPN, PRINT_PPN, ONLINE_ISSN, PRINT_ISSN, EXPECTED_LANGUAGES, ENTRY_POINT_URL, UPLOAD_OPERATION, UPDATE_WINDOW.\n"
+              << "\t                                Ignored when importing entries (all importable fields will be imported).\n\n";
+    std::exit(EXIT_FAILURE);
+}
+
+
+struct CommandLineArgs {
+    enum Mode { INVALID, IMPORT, UPDATE };
+
+    bool overwrite_on_import_;
+    std::string config_path_;
+    Mode mode_;
+    Zeder::Flavour zeder_flavour_;
+    std::set<unsigned> zeder_ids_;
+    std::set<Config::JournalParams::IniKey> fields_to_update_;
+public:
+    explicit CommandLineArgs() : overwrite_on_import_(false), mode_(INVALID) {}
+};
+
+
+void ParseCommandLineArgs(int * const argc, char *** const argv, CommandLineArgs * const commandline_args) {
```

Comment:
Why are you you passing in the addresses of the parameters to main instead of a const copy for argc and a const reference for argv?

Reply:
Because I modify those values while parsing the args.